search_knn.go

```diff
@@ -351,10 +354,10 @@ func validateAndDistributeKNNHits(knnHits []*search.DocumentMatch, indexes []Ind
 	segregatedKnnHits := make(map[string][]*search.DocumentMatch)
 	for _, hit := range knnHits {
 		if len(hit.IndexNames) == 0 {
-			return nil, ErrorPreSearchFailed
+			return nil, ErrorSearchInconsistency
```

Comment:
ErrorPreSearchFailed is not used anywhere so i just removed it

Reply:
no 
its an `or` condition, 
if `len(hit.IndexNames) == 0` then topology change occurred that replaced an alias with another
the presearch / phase 1 would have been completed by this point

Reply:
My comment is more around the cause for these 2 error situations being different - which warrant different error messaging - do you disagree?

How about you add some commentary around when `len(hit.IndexNames) is 0` and when line:359 below can occur. Perhaps that'll clarify things.

Reply:
ok yea the stack wont be empty in any topology change scene. 
but since i am peeking the stack top just to be safe im checking if stack is empty just to avoid panik